src/debug.c

```diff
@@ -32,6 +32,10 @@ void __time_critical_func(debug_printf)(const char *format, ...) {
     vsnprintf(buf, sizeof(buf), format, args);
     va_end(args);
 
+#if DEBUG_USB_UART
+    printf("%s", buf);
```

Comment:
I think that's better to go into debug_task - so that whatever the printf does doesn't block the simulation core

Reply:
Agreed!